### PR TITLE
Tweaks how some tongues have no taste

### PR DESCRIPTION
**Pull request by coiax**
Skeletons, abductors and ethereals have no sense of taste. Previously
they would taste something "indescribable", but instead, they will not
taste anything or get any message. This also means they will no longer
get mood buffs from eating/drinking high quality food.

:cl: coiax
tweak: Skeletons, abductors, etherals and anyone without a tongue
can no longer taste anything, or get food/drink moodlets.
tweak: People with the ageusia quirk will also not get any taste
messages, or get food/drink moodlets.
/:cl:

- The utility item "taster" has had some additional messages added.
- Adds a name and description to snail tongues, and colours them
  brown in leu of a proper sprite.

### CODE REVIEW

**Comment by RaveRadbury** [comment]
Is this being applied to ageusia as well?

**Comment by coiax** [comment]
Yes, people with ageusia will also not taste anything.

File: code/modules/mob/living/taste.dm [hunk 1]
```diff
@@ -4,21 +4,31 @@
...
 /mob/living/proc/taste(datum/reagents/from)
+	if(HAS_TRAIT(src, TRAIT_AGEUSIA))
+		return
+
+	var/taste_sensitivity = get_taste_sensitivity()
+
 	if(last_taste_time + 50 < world.time)
-		var/taste_sensitivity = get_taste_sensitivity()
-		var/text_output = from.generate_taste_message(taste_sensitivity,src)
+		var/text_output = from.generate_taste_message(src, taste_sensitivity)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/taste.dm:31]: Why call this before the cooldown?

File: code/modules/reagents/chemistry/holder.dm [hunk 2]
```diff
@@ -924,8 +924,9 @@
  *
  * Arguments:
  * * minimum_percent - the lower the minimum percent, the more sensitive the message is.
+ * * mob/living/taster - who is doing the tasting. Some mobs can pick up specific flavours.
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/holder.dm:927]: You should make the order of the args match the comment, or vis versua

File: code/modules/surgery/organs/tongue.dm [hunk 3]
```diff
@@ -8,6 +8,11 @@
 	attack_verb_simple = list("lick", "slobber", "slap", "french", "tongue")
 	var/list/languages_possible
 	var/say_mod = null
+	/*
+	* Whether the owner of this tongue can taste anything. Being
+	* set to FALSE will mean no taste feedback will be provided.
+	*/
```
Comment (LemonInTheDark) [inline code/modules/surgery/organs/tongue.dm:14]: This isn't the proper format for autodoc comments

File: code/modules/mob/living/taste.dm [hunk 4]
```diff
@@ -4,21 +4,31 @@
 	var/last_taste_time
 	var/last_taste_text
 
+/*
+* Gets the "taste_sensitivity" of a given mob. This is used in calculating
+* what flavours the mob can pick up, with a lower number closer to 0
+* being better.
+*/
```
Comment (LemonInTheDark) [inline code/modules/mob/living/taste.dm:11]: Nor is this I don't think

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Do you have maint approval for the species changes?

File: code/modules/mob/living/taste.dm (hunk 1 above)
Comment (coiax) [reply code/modules/mob/living/taste.dm:31]: Do you mean why call `get_taste_sensitivity()` before the cooldown check?

Comment (LemonInTheDark) [reply code/modules/mob/living/taste.dm:31]: yes

File: code/modules/surgery/organs/tongue.dm (hunk 3 above)
Comment (coiax) [reply code/modules/surgery/organs/tongue.dm:14]: Is there a method of doing DMDOC multiline variable comments, or do I just use a series of `///` comments?

Comment (LemonInTheDark) [reply code/modules/surgery/organs/tongue.dm:14]: I'm honestly not sure, I believe a series of `///` will break, one liners are best, but if more then one is needed your best bet is the proc comment format

File: code/modules/mob/living/taste.dm [hunk 5]
```diff
@@ -4,21 +4,32 @@
 	var/last_taste_time
 	var/last_taste_text
 
+/*
+ * Gets taste sensitivity of given mob
+ *
+ * This is used in calculating what flavours the mob can pick up,
+ * with a lower number being able to pick up more distinct flavours.
+ */
```
Comment (LemonInTheDark) [inline code/modules/mob/living/taste.dm:12]:
```suggestion
/**
 * Gets taste sensitivity of given mob
 *
 * This is used in calculating what flavours the mob can pick up,
 * with a lower number being able to pick up more distinct flavours.
 */
```

**Comment by Learner10** [comment]
As an ethereal main, this bothers me, especially as mine was often a detective who'd enjoy a donut and a drink, if at least for the sizzle and texture. I was never able to explain the flavor of things, only that I just liked them, which I was ok with.

**Pull request merged**